05/main.js

```diff
@@ -13,7 +13,11 @@ Student.prototype.addGrade = function (subject, grade) {
 Student.prototype.calcAverageGrade = function (subject) {
 	if (typeof subject !== 'undefined') {
 		const average = this.grades[subject].reduce((a, b) => a + b, 0) / this.grades[subject].length
-		return average
+		if (average === 0) {
+			console.log(`Brak ocen z przedmiotu ${subject}`)
```

Comment:
Niestety jest to zrobione w złym miejscu - musimy sprawdzić przed dzieleniem czy teki przedmiot występuje czyli zaraz za if zrobić warunek. tj.
```
if (typeof subject !== 'undefined') {
  if(this.grades[subject].length > 0) {
    ...
  }
}
```

PS. Obecnie trudno sobie to wyobrazić bo zawsze przy dodawaniu przedmiotu dodajemy ocenę, ale program może się rozbudować i warto już mieć takie zabezpieczenie